<launch>
    
    <!-- Arguments, can be passed in using command line -->
    <arg name="waypoints_file"      default="waypoints_1m.csv"  doc="waypoints file name" />
    <arg name="lookahead_distance"  default="5"                doc="lookahead distance" />
    <arg name="lanelet2_map_name"  default="tartu_demo.osm"    doc="lanelet2 map name" />
    <arg name="speed_limit"  default="40" doc="speed limit" />

    <!-- Nodes -->

    <!-- Waypoint loader -->
    <node pkg="practice_4" type="lanelet2_global_planner.py" name="lanelet2_global_planner" output="screen" required="true" >
        <param name="lanelet2_map_name"    value="$(find common)/data/maps/$(arg lanelet2_map_name)" />
        <param name="speed_limit"    value="$(arg speed_limit)" />
    </node>

    <!-- Global path visualizer -->
    <node pkg="practice_4" type="global_path_visualizer.py" name="global_path_visualizer" output="screen" required="true" />

    <!-- Bicycle_model simulator -->
    <node pkg="practice_4" type="bicycle_simulation.py" name="bicycle_simulation" output="screen" required="true" />

    <!-- Waypoint follower -->
    <node pkg="practice_4" type="pure_pursuit_follower.py" name="pure_pursuit_follower" output="screen" required="true" >
        <param name="lookahead_distance"  value="$(arg lookahead_distance)" />
        <remap from="path" to="global_path"/>
    </node>

    <node pkg="practice_4" type="lanelet2_map_visualizer.py" name="lanelet2_map_visualizer" output="screen" required="true" >
        <param name="lanelet2_map_name"  value="$(find common)/data/maps/$(arg lanelet2_map_name)" />
    </node>

    <!-- Rviz -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find practice_4)/rviz/practice_4.rviz" required="true" />

    <!-- Config -->
    <rosparam command="load" file="$(find practice_4)/config/localization.yaml" ns="localization" />
    <rosparam command="load" file="$(find practice_4)/config/planning.yaml" />
    <rosparam command="load" file="$(find practice_4)/config/vehicle.yaml" />

</launch>
